stripe/src/main/java/com/stripe/android/view/CountryTextInputLayout.kt

```diff
@@ -148,6 +150,36 @@ internal class CountryTextInputLayout @JvmOverloads constructor(
         }
     }
 
+    override fun onSaveInstanceState(): Parcelable? {
+        selectedCountry?.let {
+            return CountryTextInputLayoutState(
+                it.code.value,
+                super.onSaveInstanceState()
+            )
+        } ?: run {
+            return super.onSaveInstanceState()
+        }
+    }
+
+    override fun onRestoreInstanceState(state: Parcelable?) {
+        if (state is CountryTextInputLayoutState) {
+            super.onRestoreInstanceState(state.superState)
+            CountryCode.create(state.countryCode).let { countryCode ->
+                updatedSelectedCountryCode(countryCode)
+                updateUiForCountryEntered(countryCode)
+                requestLayout()
+            }
+        } else {
+            super.onRestoreInstanceState(state)
+        }
+    }
+
+    @Parcelize
+    data class CountryTextInputLayoutState(
+        val countryCode: String,
+        val superState: Parcelable?
+    ) : Parcelable
```

Comment:
Can you move this to the bottom of the class to separate it from methods? Also maybe `SelectedCountryState` Also should we be using `CountryCode` here?

(+ @michelleb-stripe)

Reply:
Done with `CountryCode` Parcelized